### PR TITLE
marsRover55: Require being logged in to access the rover page. Should redirect to login if not logged in.

### PR DESCRIPTION
**Pull request by HeatherJLewis**
**Description**
Require being logged in to access the rover page. Should redirect to login if not logged in.

**Changes**
* Refactors code to tidy up routes
* Adds new user route which is wrapped in passport.authenticate()
* Redirects to login if the authenticate fails
* Redirects to the homepage after login

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,6 +34,7 @@
     "cookie-parser": "^1.4.5",
     "express": "^4.17.1",
     "jsonwebtoken": "^8.5.1",
+    "jwt-decode": "^3.1.2",
```
Comment (Kenneth-Siu) [inline package.json:37]: This has been added but I don't see any actual references to this. Am I missing something?

File: app/homepage/index.js [hunk 2]
```diff
@@ -66,7 +66,7 @@ const getPhotoOfTheDay = () => {
 };
 
 const getUsername = () => {
-	fetch('/getUsername')
+	fetch('/user/getUsername')
```
Comment (Kenneth-Siu) [inline app/homepage/index.js:69]: This is fine for now! We might talk a bit about REST APIs in the future.

File: package.json (hunk 1 above)
Comment (HeatherJLewis) [reply package.json:37]: I think I might have accidentally deleted this during the rebase. The code that uses this is in #30 .
I've added it again as it was missing.

Comment (HeatherJLewis) [reply package.json:37]: Yes, looking back through the commits, I must have deleted it somewhere or done something else to it but I've no idea where that error must have occurred.
This adds `jwt-decode` back in.

**Pull request merged**